### ISSUE TITLE
"import/extensions": ignorePackages value does not inforce the use of extensions.

### ISSUE THREAD

**Issue by fthebaud** [benmosher/eslint-plugin-import#1083]
I'm using **eslint 4.19.1** and **eslint-plugin-import 2.11.0** with the following configuration:

```
settings: {
    react: {
      version: "16.3.0"
    },
    'import/extensions': ['.js', '.jsx'],
    'import/resolver': {
      webpack : {
        config: path.join(__dirname, '/config/webpack.config.dev.js')
      }
    },
  },
'import/extensions': ['warn', 'ignorePackages'],
```

The plugin does not raise any warning when I import files without extensions.

To reproduce the issue you can clone the project below and look for the file **TestImportWithWarnings.jsx**. https://github.com/fthebaud/react-boilerplate

**Comment by ljharb** [comment]
@fthebaud what happens if you configure the rule with `['warn', 'always', 'ignorePackages']` or `['warn', 'never', 'ignorePackages']`?

**Comment by fthebaud** [comment]
@ljharb I tested the following configurations:

`'import/extensions': ['warn', 'always']`:
I get warnings for missing extensions, but even on `import React from 'react';` (This is the reason I want "ignorePackages")

`'import/extensions': ['warn', 'always', 'ignorePackages']`:
 I don't get any warnings any more.

` 'import/extensions': ['warn', 'never']`:
I get warnings when using .js and .jsx extensions, but not for .scss extensions.

`'import/extensions': ['warn', 'never', 'ignorePackages']`:
 I don't get any warnings any more.

**Comment by collinsauve** [comment]
@fthebaud I downloaded that repo.  I believe this is just a confusion in how the options for `import/extension` works.   The [documentation here](https://github.com/benmosher/eslint-plugin-import/blob/8def087527ff4e392a45f09b7267031e9c7db857/docs/rules/extensions.md) is either missing some instructions or might just be plain incorrect.

In your repo, these options will enforce that all imports are done with extensions *except* package imports.   I believe this is what you are looking for?

```
"import/extensions": [
    "warn",
    "always",
    {
      "ignorePackages": true
    }
], 
```

@ljharb [your comment on the PR](https://github.com/benmosher/eslint-plugin-import/pull/827#issuecomment-300620018) is probably the best explanation out there as to how these options currently work.

**Comment by millerized** [comment]
>I tried the suggested configuration: 'import/extensions': ['warn', 'always', { 'ignorePackages': true }], but it's not working and I get an error:

@fthebaud: That error looks like an old version of import/exts. is being loaded. Perhaps trying blowing away `node_modules` and re-installing?

I use `['error', 'always', { 'ignorePackages': true }]` daily and it is working great.  I'll try out your repo once I get home later and see if I can reproduce on my end.

**Comment by fthebaud** [comment]
@ljharb Thanks, I'll be more careful in the future!
